### PR TITLE
Added log-in system (first half)

### PR DESCRIPTION
**Pull request by awqfrxx**
**Together with Koen we added:**

- Ability to log in
- Ability to log out
- Register users that are put into the database
- Hide admin panel from non-logged in users

**To-Do:**

- Make two different users because right now every logged in user is an admin and can see the panel (researcher account vs admin account)
- Email verification does not work
- 'Forgot your password' doesn't work because of the reason above
- 'Remember me' checkbox doesn't work

### CODE REVIEW

**Comment by awqfrxx** [comment]
The in-line styling is not easy to remove because we use 3 different css files and it's hard to overwrite.
If someone has tips or knows how to feel to comment

**Also it might be hard to understand a lot of the code, so try to pull my branch and check it out on your local machine and see how it looks/works!**

File: app/Http/Controllers/Auth/AuthenticatedSessionController.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Handle an incoming authentication request.
+     *
+     * @param  \App\Http\Requests\Auth\LoginRequest  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(LoginRequest $request)
+    {
+        $request->authenticate();
+
```
Comment (avanheeren) [inline app/Http/Controllers/Auth/AuthenticatedSessionController.php]: whitespace

File: app/Http/Controllers/Auth/AuthenticatedSessionController.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        $request->session()->regenerate();
+
```
Comment (avanheeren) [inline app/Http/Controllers/Auth/AuthenticatedSessionController.php:33]: whitespace

File: app/Http/Controllers/Auth/AuthenticatedSessionController.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function destroy(Request $request)
+    {
+        Auth::guard('web')->logout();
+
+        $request->session()->invalidate();
+
+        $request->session()->regenerateToken();
+
```
Comment (avanheeren) [inline app/Http/Controllers/Auth/AuthenticatedSessionController.php:49]: whitespace

File: app/Http/Controllers/Auth/EmailVerificationPromptController.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+     * Display the email verification prompt.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return mixed
+     */
+    public function __invoke(Request $request)
+    {
+        return $request->user()->hasVerifiedEmail()
+                    ? redirect()->intended(RouteServiceProvider::HOME)
+                    : view('auth.verify-email');
```
Comment (avanheeren) [inline app/Http/Controllers/Auth/EmailVerificationPromptController.php:21]: one indent too much?

File: app/Http/Controllers/Auth/NewPasswordController.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+                event(new PasswordReset($user));
+            }
+        );
+
+        // If the password was successfully reset, we will redirect the user back to
+        // the application's home authenticated view. If there is an error we can
+        // redirect them back to where they came from with their error message.
+        return $status == Password::PASSWORD_RESET
+                    ? redirect()->route('login')->with('status', __($status))
```
Comment (avanheeren) [inline app/Http/Controllers/Auth/NewPasswordController.php:59]: indent too much?

File: app/Http/Controllers/Auth/PasswordResetLinkController.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        // to send the link, we will examine the response then see the message we
+        // need to show to the user. Finally, we'll send out a proper response.
+        $status = Password::sendResetLink(
+            $request->only('email')
+        );
+
+        return $status == Password::RESET_LINK_SENT
+                    ? back()->with('status', __($status))
+                    : back()->withInput($request->only('email'))
+                            ->withErrors(['email' => __($status)]);
```
Comment (avanheeren) [inline app/Http/Controllers/Auth/PasswordResetLinkController.php:45]: indenting

File: app/Http/Requests/Auth/LoginRequest.php [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+     *
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function authenticate()
+    {
+        $this->ensureIsNotRateLimited();
+
+        if (! Auth::attempt($this->only('email', 'password'), $this->filled('remember'))) {
+            RateLimiter::hit($this->throttleKey());
+
```
Comment (avanheeren) [inline app/Http/Requests/Auth/LoginRequest.php:50]: whitespace

File: tests/Feature/RegistrationTest.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+use Tests\TestCase;
+
+class RegistrationTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function test_registration_screen_can_be_rendered()
+    {
+        $response = $this->get('/register');
+
```
Comment (avanheeren) [inline tests/Feature/RegistrationTest.php]: whitedspace

File: tests/Feature/PasswordResetTest.php [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+        $user = User::factory()->create();
+
+        $this->post('/forgot-password', ['email' => $user->email]);
+
+        Notification::assertSentTo($user, ResetPassword::class, function ($notification) {
+            $response = $this->get('/reset-password/'.$notification->token);
+
+            $response->assertStatus(200);
+
```
Comment (avanheeren) [inline tests/Feature/PasswordResetTest.php:45]: whitespace

File: routes/auth.php [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+Route::get('/confirm-password', [ConfirmablePasswordController::class, 'show'])
+                ->middleware('auth')
+                ->name('password.confirm');
+
+Route::post('/confirm-password', [ConfirmablePasswordController::class, 'store'])
+                ->middleware('auth');
+
+Route::post('/logout', [AuthenticatedSessionController::class, 'destroy'])
+                ->middleware('auth')
+                ->name('logout');
```
Comment (avanheeren) [inline routes/auth.php:64]: indenting

File: resources/views/welcome.blade.php [hunk 11]
```diff
@@ -11,6 +11,12 @@
         <div class="container">
             <div class="columns">
                 <div class="card">
+                    @auth()
```
Comment (avanheeren) [inline resources/views/welcome.blade.php:14]: add whitespace here or remove on 19

File: resources/views/dashboard.blade.php [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+{{--    <div class="py-12">--}}
+{{--        <div class="max-w-7xl mx-auto sm:px-6 lg:px-8">--}}
+{{--            <div class="bg-white overflow-hidden shadow-sm sm:rounded-lg">--}}
+{{--                <div class="p-6 bg-white border-b border-gray-200">--}}
+{{--                    You're logged in!--}}
+{{--                </div>--}}
+{{--            </div>--}}
+{{--        </div>--}}
+{{--    </div>--}}
+{{--</x-app-layout>--}}
```
Comment (avanheeren) [inline resources/views/dashboard.blade.php]: do the comments have to be here?

File: resources/views/auth/register.blade.php [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+                <x-input id="email" class="block mt-1 w-full" type="email" name="email" :value="old('email')" required />
+            </div>
+
+            <!-- Password -->
+            <div class="mt-4">
+                <x-label for="password" :value="__('Password')" />
+
+                <x-input id="password" class="block mt-1 w-full"
+                                type="password"
+                                name="password"
```
Comment (avanheeren) [inline resources/views/auth/register.blade.php:43]: indenting

File: resources/views/auth/register.blade.php [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+                                required autocomplete="new-password" />
+            </div>
+
+            <!-- Confirm Password -->
+            <div class="mt-4">
+                <x-label for="password_confirmation" :value="__('Confirm Password')" />
+
+                <x-input id="password_confirmation" class="block mt-1 w-full"
+                                type="password"
+                                name="password_confirmation" required />
```
Comment (avanheeren) [inline resources/views/auth/register.blade.php:53]: indenting

File: resources/views/auth/register.blade.php [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+            <div>
+                <x-label for="name" :value="__('Name')" />
+
+                <x-input id="name" class="block mt-1 w-full" type="text" name="name" :value="old('name')" required autofocus />
+            </div>
+
+            <!-- Email Address -->
+            <div class="mt-4">
+                <x-label for="email" :value="__('Email')" />
+
```
Comment (avanheeren) [inline resources/views/auth/register.blade.php:33]: whitespace

File: resources/views/auth/login.blade.php [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+            </div>
+
+            <!-- Password -->
+            <div class="mt-4">
+                <x-label for="password" :value="__('Password')" />
+
+                <x-input id="password" class="block mt-1 w-full"
+                                type="password"
+                                name="password"
+                                required autocomplete="current-password" />
```
Comment (avanheeren) [inline resources/views/auth/login.blade.php:32]: indent

File: resources/views/auth/login.blade.php [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+        <!-- Validation Errors -->
+        <x-auth-validation-errors class="mb-4" :errors="$errors" />
+
+        <form method="POST" action="{{ route('login') }}">
+            @csrf
+
+            <!-- Email Address -->
+            <div>
+                <x-label for="email" :value="__('Email')" />
+
```
Comment (avanheeren) [inline resources/views/auth/login.blade.php:21]: whitespace

File: resources/views/auth/login.blade.php [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+            <x-slot name="logo">
+            </x-slot>
+                {{--            <a href="/">--}}
+                {{--                <x-application-logo class="w-20 h-20 fill-current text-gray-500" />--}}
+                {{--                <section class="hero  is-medium  is-bold">--}}
+                {{--                    <div class="header-img">--}}
+                {{--                        <img src="./image/sdg-header-en.png" />--}}
+                {{--                    </div>--}}
+                {{--                </section>--}}
+                {{--            </a>--}}
```
Comment (avanheeren) [inline resources/views/auth/login.blade.php]: comments?

File: resources/views/auth/confirm-password.blade.php [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+            @csrf
+
+            <!-- Password -->
+            <div>
+                <x-label for="password" :value="__('Password')" />
+
+                <x-input id="password" class="block mt-1 w-full"
+                                type="password"
+                                name="password"
+                                required autocomplete="current-password" />
```
Comment (avanheeren) [inline resources/views/auth/confirm-password.blade.php:26]: indents

File: resources/views/admin.blade.php [hunk 20]
```diff
@@ -98,7 +98,7 @@
                                                 <td>{{$researchGroup->name}}</td>
                                                 <td>
                                                     <button class="has-text-right">
-                                                        <a href="{{ route('research_groups.edit', $researchGroup) }}">Edit</a>
+                                                         <a href="{{ route('research_groups.edit', $researchGroup) }}">Edit</a>
```
Comment (avanheeren) [inline resources/views/admin.blade.php]: one space too much

**Review by avanheeren — CHANGES_REQUESTED** [review]
I can see you've been working hard, nice job, just some teeny tiny things like whitespace and indenting

File: app/Http/Requests/Auth/LoginRequest.php (hunk 7 above)
Comment (awqfrxx) [reply app/Http/Requests/Auth/LoginRequest.php:50]: We keep it like this ,2 different classes being called

File: routes/auth.php (hunk 10 above)
Comment (awqfrxx) [reply routes/auth.php:64]: I disagree, the arrows are good next to eachother

**Review by Koreyoshi159 — APPROVED** [review]
Its fine

File: app/Http/Controllers/Auth/EmailVerificationPromptController.php (hunk 4 above)
Comment (awqfrxx) [reply app/Http/Controllers/Auth/EmailVerificationPromptController.php:21]: no

File: app/Http/Controllers/Auth/NewPasswordController.php (hunk 5 above)
Comment (awqfrxx) [reply app/Http/Controllers/Auth/NewPasswordController.php:59]: no

File: tests/Feature/PasswordResetTest.php (hunk 9 above)
Comment (awqfrxx) [reply tests/Feature/PasswordResetTest.php:45]: no

**Pull request merged**